File: fragment_java/app/src/main/java/com/example/fragment_java/ViewPagerAdapter.java
package com.example.fragment_java;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;

public class ViewPagerAdapter extends FragmentStateAdapter {

    public ViewPagerAdapter(@NonNull FragmentActivity fragmentActivity) {
        super(fragmentActivity);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        if (position == 0){
            return new FirstFragment();
        }
        else if (position == 1){
            return new SecondFragment();
        }
        else {
            return new ThirdFragment();
        }
    }

    @Override
    public int getItemCount() {
        return 3;
    }

}

File: fragment_kotlin/app/src/main/java/com/example/fragment_kotlin/ViewPagerAdapter.kt
package com.example.fragment_kotlin

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.example.fragment_kotlin.FirstFragment
import com.example.fragment_kotlin.SecondFragment
import com.example.fragment_kotlin.ThirdFragment

class ViewPagerAdapter(fragmentActivity: FragmentActivity) :
    FragmentStateAdapter(fragmentActivity) {
    override fun createFragment(position: Int): Fragment {
        return if (position == 0) {
            FirstFragment()
        } else if (position == 1) {
            SecondFragment()
        } else {
            ThirdFragment()
        }
    }

    override fun getItemCount(): Int {
        return 3
    }
}